### ISSUE TITLE
VMAlert should have remoteWriteSettings and remoteRead headers support

### ISSUE THREAD

**Issue by hellorill** [VictoriaMetrics/operator#492]
We use M3 as a persistent multi-tenant storage, where tenants are implemented by using some specific headers to filter metrics by labels or change metric labels before writing to the storage, for example for queries:
`M3-Restrict-By-Tags-JSON: '{"match":[{"name":"remotewrite_source","type":"EQUAL","value":"zone1"}],"strip":["remotewrite_source"]}'`

VMAgent supports remoteWriteSettings, so we can add additional labels before sending metrics to the remoteWrite storage. VMAlert does not, so for M3 it manages alert/rule states globally. We need to setup an additional proxy to add headers on read/write operations, which is not very flexible for multiple VMAlert instances.

If it's helpful, we can make changes by our team and submit them for review.

**Comment by hellorill** [comment]
Unfortunately it didn't work. VMAlert sections for remoteRead/remoteWrite:
```
  remoteWrite:
    url: "http://m3coordinator.example.com/api/v1/prom/remote/write?extra_label=remotewrite_source=test"
    concurrency: 4
  remoteRead:
    url: "http://m3query.example.com?extra_label=remotewrite_source=test"
```

When querying ALERTS and ALERTS_FOR_STATE from m3query, remotewrite_source tags are missing from labels:
```
ALERTS{alertgroup="test-alerts", alertname="vmalert always-on test alert", alertstate="firing", job="<no value>", severity="test"}
ALERTS_FOR_STATE{alertgroup="test-alerts", alertname="vmalert always-on test alert", job="<no value>", severity="test"}
```

**Comment by hellorill** [comment]
Found solution for remoteWrtie in the documentation: externalLabels, it works, but I'm not sure, that remoteRead uses those labels to limit alert requests.

**Comment by hellorill** [comment]
Checked in the code, externalLabels is loaded at startup in the manager and used in rr/rw. Also, checked Alertmanager, multiple instances of VMAlert using different alerts from the remoteRead, filtering it by externalLabels.
The current issue is no longer relevant, thanks!

**Comment by f41gh7** [comment]
Was fixed at v0.27 release, feel free to re-open if issue wasn't solved.
